scripts/test_images.sh

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+IFS=$'\n\t'
+
+source ./scripts/test_lib.sh
+source ./scripts/build_lib.sh
+
+# Test version
+cmdArray=("etcd --version" "etcdctl version" "etcdutl version")
+
+function startContainer {
+    # run docker in the background
+    docker run -d --rm --name "${RUN_NAME}" "${IMAGE}" &
+
+    # wait for etcd daemon to bootstrap
+    sleep 5
+}
+
+function stopContainer {
+    docker stop "${RUN_NAME}"
+}
+
+# Can't proceed without docker
+if ! command -v docker >/dev/null; then
+    log_error "cannot find docker"
+    exit 1
+fi
+
+# You can't run darwin binaries in linux containers
+if [[ $(go env GOOS) == "darwin" ]]; then
+    echo "Please use linux machine for release builds."
+    exit 1
+fi
+
+# Docker build
+ARCH=$(go env GOARCH)
+DOCKERFILE="Dockerfile-release.${ARCH}"
+BINARYDIR=${BINARYDIR:-"bin"}
+# Pick defaults based on release workflow
+REPOSITARY=${REPOSITARY:-"gcr.io/etcd-development/etcd"}
+if [ -n "$VERSION" ]; then
+    # Expected Format: v3.6.99-amd64
+    TAG=v"${VERSION}"-"${ARCH}"
+else
+    echo "Terminating test, VERSION not supplied"
+    exit 1
+fi
+IMAGE=${IMAGE:-"${REPOSITARY}:${TAG}"}
+
+# ETCD related values
+RUN_NAME="test_etcd"
+KEY="foo"
+VALUE="bar"
+
+# Build only if image is not present
+if [[ "$(docker images -q "${IMAGE}" 2> /dev/null)" == "" ]]; then
+    echo "${TAG} not present locally, building it..."
+
+    # Build if binaries are not present
+    if [ -z "$BINARYDIR" ]; then
+        run ./scripts/build.sh
+    fi
+
+    # Build a local image from bin directory
+    if ! docker build -t "${IMAGE}" -f "${DOCKERFILE}" "${BINARYDIR}"; then
+        echo "Docker build unsuccessful. Exit code $?"
+        exit 1
+    fi
+fi
+
+startContainer
+
+# Version check
+# set IFS to include space so that docker exec command does not consider
+# arguments as part of exec name. This prevents the need to maintain
+# an additonal array.
+IFS=$'\n\t '
+for i in "${!cmdArray[@]}"; do
+    Cmd=("/usr/local/bin/""${cmdArray[$i]}")
+    # shellcheck disable=SC2068
+    Out=$(docker exec "${RUN_NAME}" ${Cmd[@]})
+    foundVersion=$(echo "$Out" | head -1 | cut -f3 -d" ")
```

Comment:
`-f3` is a little fragile. Let's just get the last field.
```suggestion
    foundVersion=$(echo "$Out" | head -1 | rev | cut -d' ' -f 1 | rev)
```

Reply:
```
docker exec etcd etcd --version | head -1 | rev |cut -d' ' -f 1 |rev
3.5.7
```

Since the version (e.g. "3.5.7") is always at the end of the first line, so getting the last field is better than get 3rd field. It avoid depending on a specific position/index (3 in this case). 

Of course, this is just a minor comment.

Reply:
My bad, I just checked the etcd output and it prints in separate lines. I will incorporate in next updated.